### PR TITLE
Implement "no changes requested" merge policy

### PR DESCRIPTION
**Pull request by mmitche**
Adds a merge policy which disallows merges if changes are requested or the PR is rejected altogether

### CODE REVIEW

**Comment by mmitche** [comment]
@natemcmaster Fixes https://github.com/dotnet/arcade/issues/1999

File: src/Maestro/Maestro.MergePolicies/NoRequestedChangesMergePolicy.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+{
+    public class NoRequestedChangesMergePolicy : MergePolicy
+    {
+        public override string DisplayName => "No Requested Changes";
+
+        public override async Task EvaluateAsync(IMergePolicyEvaluationContext context, MergePolicyProperties properties)
+        {
+            IEnumerable<Review> reviews = await context.Darc.GetPullRequestReviewsAsync(context.PullRequestUrl);
+
+            // Look for any reviews that are "rejected" or "rejected"
```
Comment (alexperovich) [inline src/Maestro/Maestro.MergePolicies/NoRequestedChangesMergePolicy.cs]: You said "rejected" twice
```suggestion
            // Look for any reviews that are "rejected"
```

File: src/Microsoft.DotNet.Darc/src/DarcLib.AzDev/AzureDevOpsClient.cs [hunk 2]
```diff
@@ -675,6 +675,58 @@ public async Task<IList<Check>> GetPullRequestChecksAsync(string pullRequestUrl)
...
+            IList<Review> reviews = new List<Review>();
+            foreach (JToken review in values)
+            {
+                int vote = review["vote"].Value<int>();
+
+                ReviewState reviewState;
+
+                switch (vote)
+                {
+                    case 10:
```
Comment (alexperovich) [inline src/Microsoft.DotNet.Darc/src/DarcLib.AzDev/AzureDevOpsClient.cs]: These are a bunch of magic numbers. Add something that explains what they mean.

Comment (mmitche) [reply src/Microsoft.DotNet.Darc/src/DarcLib.AzDev/AzureDevOpsClient.cs]: I'll link to the azdo documentation.  They are in fact...a bunch of magic numbers.

File: src/Microsoft.DotNet.Darc/src/DarcLib/Actions/Remote.cs [hunk 3]
```diff
@@ -250,13 +250,31 @@ public async Task DeleteBranchAsync(string repoUri, string branch)
             }
         }
 
+        /// <summary>
+        ///     Get a list of pull request checks.
+        /// </summary>
+        /// <param name="pullRequestUrl">Url of pull requests</param>
```
Comment (riarenas) [inline src/Microsoft.DotNet.Darc/src/DarcLib/Actions/Remote.cs]:
```suggestion
        /// <param name="pullRequestUrl">Url of pull request</param>
```

File: src/Microsoft.DotNet.Darc/src/DarcLib/Actions/Remote.cs [hunk 4]
```diff
@@ -250,13 +250,31 @@ public async Task DeleteBranchAsync(string repoUri, string branch)
...
-                IEnumerable<Check> checks = await _gitClient.GetPullRequestChecksAsync(pullRequestUrl);
-                return checks;
+                return await _gitClient.GetPullRequestChecksAsync(pullRequestUrl);
+            }
+        }
+
+        /// <summary>
+        ///     Get a list of pull request reviews.
+        /// </summary>
+        /// <param name="pullRequestUrl">Url of pull requests</param>
```
Comment (riarenas) [inline src/Microsoft.DotNet.Darc/src/DarcLib/Actions/Remote.cs]:
```suggestion
        /// <param name="pullRequestUrl">Url of pull request</param>
```

**Pull request merged**